src/pkg/caendr/caendr/models/datastore/nemascan_mapping.py

```diff
@@ -1,47 +1,70 @@
 from caendr.services.logger import logger
 
-from caendr.models.datastore import DataJobEntity
-from caendr.models.status import JobStatus
-from caendr.services.cloud.storage import check_blob_exists, get_blob_list
+from caendr.models.datastore       import HashableEntity, ReportEntity
+from caendr.models.status          import JobStatus
+from caendr.services.cloud.storage import BlobURISchema, check_blob_exists, get_blob_list
+from caendr.utils.env              import get_env_var
 
 
-NEMASCAN_REPORT_PATH_PREFIX = 'reports'
-NEMASCAN_RESULT_PATH_INFIX = 'results'
-INPUT_DATA_PATH = 'tools/nemascan/input_data'
+PRIVATE_BUCKET_NAME = get_env_var('MODULE_SITE_BUCKET_PRIVATE_NAME')
 REPORT_DATA_PREFIX = 'Reports/NemaScan_Report_'
-NEMASCAN_INPUT_FILE = 'data.tsv'
 
 
-class NemascanReport(DataJobEntity):
+
+class NemascanReport(HashableEntity, ReportEntity):
+
+  #
+  # Class Variables
+  #
+
   kind = 'nemascan_mapping'
-  _blob_prefix = NEMASCAN_REPORT_PATH_PREFIX
-  _input_file  = NEMASCAN_INPUT_FILE
 
   _report_display_name = 'Genetic Mapping'
 
-  __result_infix = NEMASCAN_RESULT_PATH_INFIX
-  __input_data_path = INPUT_DATA_PATH
-  __report_path = REPORT_DATA_PREFIX
+  # Identify the report data by the data hash, inherited from the HashableEntity parent class
+  _data_id_field = 'data_hash'
 
 
-  ## Buckets & Paths ##
+  #
+  # Paths
+  #
 
-  # TODO: Overrides function in DataJobEntity parent class which isn't needed in this class.
-  #       Can this be merged with get_result_path for better inheritance?
-  def get_result_blob_path(self):
-    raise TypeError(f'Should not call get_result_blob_path on {self.__class__.__name__}')
+  # TODO: Move data files to Data bucket
```

Comment:
We should discuss standardizing the locations of certain data files for each tool. It's come up a bit in the past, but I have a much better idea now of what files are required by each tool and how we can divide them up / access them in a regular way